orchagent/portsorch.cpp

```diff
@@ -4170,15 +4164,9 @@ void PortsOrch::doPortTask(Consumer &consumer)
                     }
                 }
 
-                if (!serdes_attr.empty())
+                if (!serdes_attr.empty() && p.m_preemphasis != serdes_attr)
```

Comment:
@stepanblyschak this will prevent setting of **SAME** serdes SI setting by OA if user updates any of the PORT table attributes in CONFIG_DB. This will _unintentionally_ prevent applying SAME SI setting when user plugs out the optics and insert it back. To avoid this we should clear/unset the p.m_preemphasis map when OA gets notified of optics removal [here](https://github.com/sonic-net/sonic-swss/blob/master/orchagent/portsorch.cpp#L4598)

Reply:
@stepanblyschak another use case where you would like to re-apply **same** SI settings is when user wants to shut the interface followed by no-shut. This should retrain the link with same SI settings. So, you should also clear the p.m_preemphasis map during admin down

Reply:
@prgeor Do you mean SONiC **must** set same serdes SI attributes if the module is re-inserted? Is it per SAI spec? What libSAI is supposed to return when user queries for serdes attributes when the module is re-inserted?
For example, there is no such requirement on NVIDIA platform.
In case of shutdown/startup - why libSAI does not handle that? For example, we don't re-set the speed/fec/autoneg parameters when the interface is admin down, then up.

Reply:
@prgeor I think clearing m_preemphasis to handle plug out event could be risky because there's no guarantee orchagent will first process STATE_TRANSCEIVER_INFO_TABLE_NAME and then APP_PORT_TABLE_NAME if both their corresponding consumers have pending data.

I am working on a different solution to make serdes config entirelly dependent on xcvrd's request.

Reply:
@stepanblyschak can you elaborate on the risk? Do you suspect OA may miss the insert/remove event of the transceiver?

Reply:
@prgeor No, the risk is that there's no guarantee orchagent will first process STATE_TRANSCEIVER_INFO_TABLE_NAME and then APP_PORT_TABLE_NAME if both consumers have data.
Then, serdes config won't be applied.

Reply:
@prgeor Please check the new appraoch.

I made the change so that while the port is not yet created, field-value pairs are aggregated in ```m_portConfigMap```, but when the port is already created only updated field-value pairs are processed.
This makes orchagent unconditionally set serdes attributes whenever xcvrd pushes the changes to APP_DB and also fixes the problem observed when mtu is configured. When mtu is configured, only mtu ```is_set``` with the new approach and serdes isn't re-configured. Serdes configuration becomes fully controlled by xcvrd and there's no special logic in orchagent for that.

Regarding admin status change I do think it is out of scope of this PR. It is not required on Nvidia platform. Even if required by SAI spec, original serdes support implementation (looking at 202205 for example) did not handle admin status change and there was no field-value pairs aggregation so it didn't re-configure serdes on every port field update including admin status change.